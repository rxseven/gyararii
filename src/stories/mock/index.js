import elements from './elements';

export default {
  elements
};
