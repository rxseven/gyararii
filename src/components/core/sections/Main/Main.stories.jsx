import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Main from './index';

const { content } = mock.elements;
const title = 'Components/Core/Sections/Main';
const component = <Main>{content}</Main>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
