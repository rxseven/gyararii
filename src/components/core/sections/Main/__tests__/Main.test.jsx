import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import mock from 'tests/mock';
import Main from '../index';

const seed = {
  props: {
    children: mock.elements.children
  }
};

describe('<Main />', () => {
  // Arrange
  const component = <Main>{seed.props.children}</Main>;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
