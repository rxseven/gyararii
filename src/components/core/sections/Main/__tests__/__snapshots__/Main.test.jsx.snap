// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Snapshot tests should render correctly 1`] = `
<main
  className="sc-bdVaJa cIUQSX"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div>
          Content
        </div>
      </div>
    </div>
  </div>
</main>
`;
